docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,80 @@ const currentUserIDState = atom({
 });
 ```
 
+### Asynchronous
+
+If your persisted data needs to be `await`ed, you can either create an `asynchronous` function and call that or use a `promise` in the setSelf function 
+
+- Using the promise approach you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback state while waiting for `Recoil` to load the persisted values.
+- This is not possible with the async await approach.
+
+#### Promise
+
+```jsx
+const localStorageEffect = key => ({setSelf, onSet}) => {
+  setSelf(
+    localStorage.getItem(key).then((savedValue) => {
+      if (savedValue != null) {
+        return JSON.parse(savedValue);
+      }
+    }),
+  );
+
+  onSet(newValue => {
+    if (newValue instanceof DefaultValue) {
+      localStorage.removeItem(key);
+    } else {
+      localStorage.setItem(key, JSON.stringify(newValue));
+    }
+  });
+};
+
+const currentUserIDState = atom({
+  key: 'CurrentUserID',
+  default: 1,
+  effects_UNSTABLE: [
+    localStorageEffect('current_user'),
+  ]
+});
+```
+
+
+#### Async Await
+
+```jsx
+
+/** If there's a persisted value - set it on load  */
+const loadPersisted = async ({ key, setSelf }) => {
```

Comment:
Passing through the `setSelf()` callback might be confusing.  To help with clarity for documentation purposes maybe define this async function in the effect function so it can just use `key` and `setSelf()` directly.

Reply:
With documentation in mind, that makes sense. I'll do that.